### DISCUSSION TITLE
Safari can't connect to secure laravel websocket

### DISCUSSION THREAD

**Discussion by Sairahcaz** [beyondcode/laravel-websockets#911] [Q&A] [answered]
I already asked this question on stackoverflow, but nobody answered:
https://stackoverflow.com/questions/70464828/safari-cant-connect-to-secure-laravel-websocket

So maybe here somebody can help me?:

I'm running an app, where I use laravel-websockets, with the pusher replacement and a self created websocket instance. For SSL I used a letsencrypt certificate and the nginx reverse proxy "[Same location for websockets and web contents][1]" approach. On my nginx TLS 1.2 and 1.3 is enabled. Now on Firefox and Chrome all is working fine, but on Safari (version 14.1.2, tested on a macOS Mojave 10.14.6) I get the following error in the console:

    WebSocket connection to 'wss://example.com/app/asdfasdf?protocol=7&client=js&version=7.0.3&flash=false' failed: Unexpected response code: 401

These are my echo options:

    const echoOptions = {
      broadcaster: 'pusher',
      key: process.env.MIX_PUSHER_APP_KEY,
      cluster: process.env.MIX_PUSHER_APP_CLUSTER,
      wsHost: window.location.hostname,
      wsPort: 80,
      wssPort: 443,
      forceTLS: true,
      enabledTransports: ['ws', 'wss'], // <-- disable pusher api fallback
      disableStats: true,
    };
    
    window.Echo = new Echo(echoOptions);

I searched a lot for any kind of similar issues, but I can't find an answer.

Any ideas or help would be awesome!

  [1]: https://beyondco.de/docs/laravel-websockets/basic-usage/ssl#same-location-for-websockets-and-web-contents

**Comment by Sairahcaz** [comment, marked as answer]
Got it working:
https://stackoverflow.com/questions/70464828/safari-cant-connect-to-secure-laravel-websocket/70669225#70669225

**Comment by erikn69** [comment]
`composer require "beyondcode/laravel-websockets:dev-master"`
With `dev-master` you have support for `pusher/pusher-php-server` 7.0
https://github.com/beyondcode/laravel-websockets/blob/fd2070fc3d28696247acceacc4f40667828f3016/composer.json#L43

**Reply by Sairahcaz** [reply]
Hmm, but I think dev-master has not really a non-changeable state? Doesn't this branch changes over time?

**Reply by erikn69** [reply]
yes, but other branches are outdated and no progress is seen

**Reply by mcribz** [reply]
fucking not works

**Reply by parallels999** [reply]
It works, thanks
